server/routes/SolutionRoutes.js

```diff
@@ -42,7 +42,21 @@ router.get("/:id", Solution.get);
 // router.get("/:id", Challenge.getComments);
 
 // Create a new solution in the database.
-router.post("/", upload.single("diagram"), Solution.create);
+const uploadMiddleware = (req, res, next) => {
```

Comment:
Why did you customize this?

Reply:
I wanted to handle both `MulterError` and other unknown errors by returning error messages to the client (400 for file upload issues and 500 for general server errors). This customization is only for Error Handling and if everything went well, the diagram will get uploaded as usual over the same route